src/connector/src/source/google_pubsub/enumerator/client.rs

```diff
@@ -53,15 +55,52 @@ impl SplitEnumerator for PubsubSplitEnumerator {
             .await
             .map_err(|e| anyhow!("error initializing pubsub client: {:?}", e))?;
 
-        if !client
-            .subscription(&subscription)
+        let sub = client.subscription(&subscription);
+        if !sub
             .exists(None, None)
             .await
             .map_err(|e| anyhow!("error checking subscription validity: {:?}", e))?
         {
             return Err(anyhow!("subscription {} does not exist", &subscription));
         }
 
+        // We need the `retain_acked_messages` configuration to be true to seek back to timestamps
+        // as done in the [`PubsubSplitReader`] and here.
+        // ? Should we enforce this policy
```

Comment:
yes, I think we should enforce it because seeking back is a regular operation in RisingWave